### PR TITLE
Gamepad binding menu and console command. Some UI fixes.

### PR DESCRIPTION
**Pull request by justtails**
What's new in this PR?
- Now you can bind gamepad buttons with console command
- You can bind gamepad buttons with UI menu in Options -> Controls
- Refactored code associated with xr_input
- Fix bug with choosing weather using ComboBox in MP -> Server menu
- Fix again misplaced weathers buttons in MP voting dialog. But without any gamedata changes

### CODE REVIEW

File: src/xrGame/ui/UIMapList.cpp [hunk 1]
```diff
@@ -240,6 +240,8 @@ void CUIMapList::LoadMapList()
 {
     const auto& weathers = gMapListHelper.GetGameWeathers();
     u32 cnt = 0;
+    if(m_pWeatherSelector->GetSize() > 0);
```
Comment (Xottab-DUTY) [inline src/xrGame/ui/UIMapList.cpp:243]: Please send it in another pull request, but also fix the missing space:
```suggestion
    if (m_pWeatherSelector->GetSize() > 0);
```

File: src/xrGame/ui/ChangeWeatherDialog.cpp [hunk 2]
```diff
@@ -86,16 +86,26 @@ void ChangeWeatherDialog::InitChangeWeather(CUIXml& xmlDoc)
     auto& gameWeathers = gMapListHelper.GetGameWeathers();
     Initialize(gameWeathers.size());
     weathers.resize(gameWeathers.size());
+    
+    // There is no mistake. Vanilla algorithm sorted config keys by alphabet.
+    // We did it for vanilla bug fixes and compataible with game data.
```
Comment (Xottab-DUTY) [inline src/xrGame/ui/ChangeWeatherDialog.cpp:91]: Can you please send this change and all related to it as a separate pull request?

File: src/xrGame/ui/UIListItemServer.cpp [hunk 3]
```diff
@@ -99,22 +99,22 @@ void CUIListItemServer::InitItemServer(LIST_SRV_ITEM& params)
 }
 
 #include "string_table.h"
-u32 cut_string_by_length(CGameFont* pFont, LPCSTR src, LPSTR dst, u32 dst_size, float length);
+u32 CutStringByLength(CGameFont* font, LPCSTR src, LPSTR dst, u32 dstSize, float length);
```
Comment (Xottab-DUTY) [inline src/xrGame/ui/UIListItemServer.cpp:102]: Not related to gamepads, not even related to input system. Please send it as another pull request for MP map weather list fix.

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]
Please, split the map weather list fix from this pull request.

File: src/xrGame/ui/UIListItemServer.cpp (hunk 3 above)
Comment (justtails) [reply src/xrGame/ui/UIListItemServer.cpp:102]: No, I can't. Because this function is the part of xr input associated code refactoring. Watch `xrGame/ui/UIEditKeyBind.cpp` and you'll see where is defenision.

**Comment by justtails** [comment]
@Xottab-DUTY Yeah, I try to split commits in different PRs, but after that you need to accept them in the correct sequence .
